### PR TITLE
Adds lists and list items and delete functionality

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This addresses #7233 and #7234.  It adds the Application List and Delete functionality for the list.  It also adds the necessary tests.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

##### ADDITIONAL INFORMATION

### CODE REVIEW

File: awx/ui_next/src/screens/Application/ApplicationsList/ApplicationsList.jsx [hunk 1]
```diff
@@ -1,15 +1,180 @@
...
+import { ApplicationsAPI } from '../../../api';
+import PaginatedDataList, {
+  ToolbarDeleteButton,
+  ToolbarAddButton,
+} from '../../../components/PaginatedDataList';
+import useSelected from '../../../util/useSelected';
+
+import ApplicationListItem from './ApplicationListItem';
+
+const QS_CONFIG = getQSConfig('inventory', {
```
Comment (jakemcdermott) [inline awx/ui_next/src/screens/Application/ApplicationsList/ApplicationsList.jsx]: I think this should be `getQSConfig('applications', {`

File: awx/ui_next/src/screens/Application/ApplicationsList/ApplicationListItem.jsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+`;
+
+function ApplicationListItem({
+  application,
+  isSelected,
+  onSelect,
+  detailUrl,
+  i18n,
+}) {
+  ApplicationListItem.propTypes = {
```
Comment (mabashian) [inline awx/ui_next/src/screens/Application/ApplicationsList/ApplicationListItem.jsx]: Normally we put this outside (underneath) the function - did we change that pattern?

Comment (AlexSCorey) [reply awx/ui_next/src/screens/Application/ApplicationsList/ApplicationListItem.jsx]: I do think I've seen it in a couple of different locations, but I agree, this should be beneath the function. At least that is where I prefer it.

**Review by mabashian — APPROVED** [review]
lgtm

**Review by jakemcdermott — APPROVED** [review]

**Comment by dsesami** [comment]
two adjustments are needed in the applications list.
12:33
1) we gotta include the "Last Modified" time and date in each item for applications
2) sort dropdown should include "last modified". "Created by" isn't a useful search here and isn't visible in the list. see the old UI for reference.

**Review by dsesami — APPROVED** [review]

**Pull request merged**